### DISCUSSION TITLE
Can not push or pull anytihng on Github Desktop and SourceTree

### DISCUSSION THREAD

**Discussion by SurpriserTR** [community/community#31931] [Other Feature Feedback, Questions, & Ideas]
https://github.com/SurpriserTR/Game-Art
I have my **private** repository where all my libraries and packages I save at.
Recently I can not access my repo tried every possible solutions on web none worked out.
![image](https://user-images.githubusercontent.com/43583369/188317045-8e74b7f1-95dc-45ac-9ae7-2b26a8efc673.png)

Even right after pulling the repo it gives this error.
There could be **corrupted** file on repository or some damaged files but I dont not how to fix it.
**Sometimes I can not undo** my commits I had  rm -rf .git command tried on several possible files but still the problem exist

**Comment by DenverCoder1** [comment]
Is `C:\Game-Art` the location it is cloned in? If not you can try using locate to tell it the correct folder.

Otherwise, you may have encountered permission issues saving in such a high-level folder on your computer.

Maybe instead try saving in your `Documents` folder, for example.

**Reply by SurpriserTR** [reply]
Locating it to the directory works but after fetching the repo it goes to Can't find tge repo again

**Comment by airtower-luna** [comment]
> Sometimes I can not undo my commits I had rm -rf .git command tried on several possible files but still the problem exist

If you delete the `.git/` directory, you *lose your local repository*. That is not something you should ever need to do. :warning: 

Now that you've already done it you have essentially two ways to recover, *if* your repository on GitHub is (mostly) up to date.

* If you don't have a lot of uncommitted & unpushed changes, do a fresh clone of the repository into a separate directory. Then copy any changed files you want to keep over. This is the easier way.
* If you can't do a fresh clone for some reason, you can manually initialize your directory as a Git repository again, add the remote repository, fetch from there, switch to the right branch, and resolve any conflicts. This is likely a lot of work.

**Reply by SurpriserTR** [reply]
I tried the first option earlier but there are so many files that I dont even remember where.
But worked thank you. I just wondered if there is a way to handle the use current directory and keep changes.

**Reply by airtower-luna** [reply]
That's the second option, yes. Basically do all the steps of a clone *except checkout* manually, and then reconcile file by file. That's why I wrote it'd likely be a lot of work. :slightly_smiling_face:
